public/app/features/folders/FolderAlerting.tsx

```diff
@@ -0,0 +1,33 @@
+import Page from 'app/core/components/Page/Page';
+import { GrafanaRouteComponentProps } from 'app/core/navigation/types';
+import { getNavModel } from 'app/core/selectors/navModel';
+import { StoreState } from 'app/types';
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { useAsync } from 'react-use';
+import { AlertsFolderView } from '../alerting/unified/AlertsFolderView';
+import { useGetFolderByUid } from './state/actions';
+import { getLoadingNav } from './state/navModel';
+
+export interface OwnProps extends GrafanaRouteComponentProps<{ uid: string }> {}
+
+const FolderAlerting = ({ match }: OwnProps) => {
+  const getFolderByUid = useGetFolderByUid();
```

Comment:
what's the reason for not using the existing action for fetching folder by uid here?

Reply:
The existing action works only for redux `connected` components. I noticed that in our components we mostly use hooks rather than `connect` HOC. I can switch to the `connect` to be aligned with the rest of the `folders` components

Reply:
Yeah but you can use `useDispatch` in this component.

```
const dispatch = useDispatch();
....
const { loading } = useAsync(async () => await dispatch(getFolderByUid(uid)), [dispatch, uid]);
```

Reply:
Yes, good point!